colibre/registration.py

```diff
@@ -249,30 +199,66 @@ def register_oxygen_to_hydrogen(self, catalogue, aperture_sizes):
-            catalogue.gas_element_ratios_times_masses,
-            f"O_over_H_times_gas_mass_{aperture_size}_kpc",
+        # Fetch O over H times gas mass computed in apertures.  The factor of 16 (the
+        # mass ratio between O and H) has already been accounted for.
+        log_O_over_H_times_gas_mass = getattr(
+            catalogue.element_ratios_times_masses,
+            f"log_O_over_H_times_gas_mass_lowfloor_{aperture_size}_kpc",
         )
         # Fetch gas mass in apertures
         gas_sf_mass = getattr(catalogue.apertures, f"mass_gas_sf_{aperture_size}_kpc")
 
         # Compute gas-mass weighted O over H
-        O_over_H = unyt.unyt_array(np.zeros_like(gas_sf_mass), "dimensionless")
+        log_O_over_H = unyt.unyt_array(np.zeros_like(gas_sf_mass), "dimensionless")
         # Avoid division by zero
         mask = gas_sf_mass > 0.0 * gas_sf_mass.units
-        O_over_H[mask] = O_over_H_times_gas_mass[mask] / gas_sf_mass[mask]
+        log_O_over_H[mask] = log_O_over_H_times_gas_mass[mask] / gas_sf_mass[mask]
 
-        # Convert to units used in observations (16 is the mass ratio between O and H)
-        O_abundance = unyt.unyt_array(12 + np.log10(O_over_H), "dimensionless")
-        O_abundance.name = f"Gas $12+\\log_{10}({{\\rm O/H}})$ ({aperture_size} kpc)"
+        # Convert to units used in observations
+        O_abundance = unyt.unyt_array(12 + log_O_over_H, "dimensionless")
+        O_abundance.name = (
+            f"SF Gas $12+\\log_{{10}}({{\\rm O/H}})$ ({aperture_size} kpc)"
+        )
 
         # Register the field
         setattr(self, f"gas_o_abundance_{aperture_size}_kpc", O_abundance)
 
     return
 
 
+def register_iron_to_hydrogen(self, catalogue, aperture_sizes, fe_solar_abundance):
+
+    # Loop over apertures
+    for aperture_size in aperture_sizes:
+
+        # Fetch Fe over H times stellar mass computed in apertures. The
+        # mass ratio between Fe and H has already been accounted for.
+        log_Fe_over_H_times_star_mass = getattr(
+            catalogue.element_ratios_times_masses,
+            f"log_Fe_over_H_times_star_mass_lowfloor_{aperture_size}_kpc",
+        )
+        # Fetch stellar mass in apertures
+        star_mass = getattr(catalogue.apertures, f"mass_star_{aperture_size}_kpc")
+
+        # Compute stellar-mass weighted Fe over H
+        Fe_over_H = unyt.unyt_array(np.zeros_like(star_mass), "dimensionless")
+        # Avoid division by zero
+        mask = star_mass > 0.0 * star_mass.units
+        Fe_over_H[mask] = pow(
+            10.0, log_Fe_over_H_times_star_mass[mask] / star_mass[mask]
+        )
+        # Convert to units used in observations
+        Fe_abundance = unyt.unyt_array(
+            np.log10(Fe_over_H / fe_solar_abundance), "dimensionless"
+        )
+        Fe_abundance.name = f"Stellar $[{{\\rm Fe/H}}]$ ({aperture_size} kpc)"
+
+        # Register the field
+        setattr(self, f"star_fe_abundance_{aperture_size}_kpc", Fe_abundance)
```

Comment:
```suggestion
    # Loop over apertures
    for aperture_size in aperture_sizes:
    
        for thr, thr_label in zip(["low", "high"], ["Min = $10^{{-4}}$", "Min = $10^{{-3}}$"]):

            # Fetch Fe over H times stellar mass computed in apertures. The
            # mass ratio between Fe and H has already been accounted for.
            log_Fe_over_H_times_star_mass = getattr(
                catalogue.element_ratios_times_masses,
                f"log_Fe_over_H_times_star_mass_{thr}floor_{aperture_size}_kpc",
            )
            # Fetch stellar mass in apertures
            star_mass = getattr(catalogue.apertures, f"mass_star_{aperture_size}_kpc")
    
            # Compute stellar-mass weighted Fe over H
            Fe_over_H = unyt.unyt_array(np.zeros_like(star_mass), "dimensionless")
            # Avoid division by zero
            mask = star_mass > 0.0 * star_mass.units
            Fe_over_H[mask] = pow(
                10.0, log_Fe_over_H_times_star_mass[mask] / star_mass[mask]
            )
            # Convert to units used in observations
            Fe_abundance = unyt.unyt_array(Fe_over_H / fe_solar_abundance, "dimensionless")
            Fe_abundance.name = f"Stellar $[{{\\rm Fe/H}}]$ ({thr_label}, {aperture_size} kpc)"
    
            # Register the field
            setattr(self, f"star_fe_abundance_{thr}_{aperture_size}_kpc", Fe_abundance)

```

Reply:
Perhaps in the above suggestion, it would be more clear to use `floor` instead of `thr`